python/ml4ir/applications/ranking/tests/test_t_test_calculation.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+from scipy import stats
+import unittest
+import warnings
+from ml4ir.base.model.relevance_model import perform_click_rank_dist_paired_t_test, compute_stats_from_stream
+
+warnings.filterwarnings("ignore")
+
+
+class TestTtestCalculation(unittest.TestCase):
+    """
+    Testing the calculation of paired t-test statistic and its p-value.
+    """
+
+    def t_test_calculation(self, a, b, batchzs, pvalue_threshold):
+        n = len(a)
+        expected_t_test_stat, expected_pvalue = stats.ttest_rel(a, b)
+        expected_decision = expected_pvalue < pvalue_threshold
+        agg_mean, agg_count, agg_M2 = 0, 0, 0
+        for bcount in range(int(n/batchzs)):
+            batch_a = a[bcount*batchzs : (bcount+1)*(batchzs)]
+            batch_b = b[bcount*batchzs : (bcount+1)*(batchzs)]
+            d = batch_a - batch_b
+            agg_count, agg_mean, agg_M2 = compute_stats_from_stream(d, agg_count, agg_mean, agg_M2)
```

Comment:
I am a bit confused here, I would need more time to go deep in your calculations, also you are doing some calculations on the `perform_click_rank_dist_paired_t_test` arguments. But in your tests, please do some sanity checks: 
- compare at least the variance analytically calculated with the one you calculate in the streamed approach
- anything else you are doing in stream vs can be done analytically
- Compare the p-values, not the decisions. I think you are using a two sided as expected and you are calculating an 1-sided in your  `perform_click_rank_dist_paired_t_test`, I need to be convinced ;-)

Reply:
Good point! I believe we should go with a two sided to be more inclusive.